### PR TITLE
perlPackages.SubStrictDecl: init at 0.005

### PR DESCRIPTION
**Pull request by pborzenkov**
###### Description of changes

The PR adds `perlPackages.SubStrictDecl` at 0.005 and its dependency that's currently missing in `nixpkgs`:
`LexicalSealRequireHints` at 0.0011

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zakame** [comment]
Result of `nixpkgs-review pr 170416` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>perl532Packages.SubStrictDecl</li>
    <li>perl534Packages.SubStrictDecl</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>perl532Packages.LexicalSealRequireHints</li>
    <li>perl534Packages.LexicalSealRequireHints</li>
  </ul>
</details>

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -20559,6 +20572,20 @@ let
...
 
+  SubStrictDecl = buildPerlModule {
+    pname = "Sub-StrictDecl";
+    version = "0.005";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/Z/ZE/ZEFRAM/Sub-StrictDecl-0.005.tar.gz";
+      sha256 = "sha256-oSfa52RcGpVwzZopcMbcST1SL/BzGKNKOeQJCY9pESU=";
+    };
+    buildInputs = [ LexicalSealRequireHints ];
+    meta = {
```
Comment (zakame) [inline pkgs/top-level/perl-packages.nix:20583]: I think we need this `perlPreHook` to fix build on Darwin (and perhaps elsewhere:)

```suggestion
    perlPreHook = lib.optionalString stdenv.isDarwin "export LD=$CC";
    meta = {
```

**Review by zakame — CHANGES_REQUESTED** [review]

**Comment by stigtsp** [comment]
@GrahamcOfBorg build perlPackages.SubStrictDecl perlPackages.LexicalSealRequireHints

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -20559,6 +20572,21 @@ let
...
   };
 
+  SubStrictDecl = buildPerlModule {
+    pname = "Sub-StrictDecl";
+    version = "0.005";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/Z/ZE/ZEFRAM/Sub-StrictDecl-0.005.tar.gz";
+      sha256 = "sha256-oSfa52RcGpVwzZopcMbcST1SL/BzGKNKOeQJCY9pESU=";
+    };
+    buildInputs = [ LexicalSealRequireHints ];
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:20582]:
```suggestion
    propagatedBuildInputs = [ LexicalSealRequireHints ];
```

Runtime dependency needs to be propagated

```
$ perl -MSub::StrictDecl
Can't locate Lexical/SealRequireHints.pm in @INC ...
```

Comment (pborzenkov) [reply pkgs/top-level/perl-packages.nix]: Yep, thank you. Fixed.

**Comment by stigtsp** [comment]
@zakame can you double check on aarch64-darwin? (ofborg is taking it's time)

**Comment by pborzenkov** [comment]
```
❯ uname -sm
Darwin arm64

❯ nix-shell -p 'perl.withPackages (p: [p.SubStrictDecl])' -I nixpkgs=.
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring

[nix-shell:~/src/nixos/nixpkgs]$ perl -MSub::StrictDecl
```

Looks good

**Review by stigtsp — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**